zephyr/wrapper.c

```diff
@@ -582,6 +584,9 @@ int task_main_start(struct sof *sof)
 #define SOF_IPC_QUEUED_DOMAIN SOF_SCHEDULE_LL_TIMER
 #endif
 
+	/* primary core enabled, keep track in wrapper */
+	w_core_enable_mask |= BIT(PLATFORM_PRIMARY_CORE_ID);
```

Comment:
locking around the RMW of the mask ?

Reply:
@lgirdwood  This is build on the assumption that this is only called from core0 and thus no locking is needed. Let me double-check and add some asserts around this to ensure this does not blow up (silently) later.

Reply:
@lgirdwood One more update, I added explicit asserts to catch any usage of these funtions from non-primary core. If we ever need such functionality, it should be done with native Zephyr interfaces instead.